site/src/modules/tasks/TaskDeleteDialog/TaskDeleteDialog.tsx

```diff
@@ -0,0 +1,58 @@
+import { API } from "api/api";
+import { getErrorDetail, getErrorMessage } from "api/errors";
+import { ConfirmDialog } from "components/Dialogs/ConfirmDialog/ConfirmDialog";
+import { displayError, displaySuccess } from "components/GlobalSnackbar/utils";
+import type { FC } from "react";
+import { QueryClient, useMutation } from "react-query";
+import type { Task } from "../tasks";
+
+type TaskDeleteDialogProps = {
+	open: boolean;
+	task: Task;
+	onClose: () => void;
+	onSuccess?: () => void;
+};
+
+export const TaskDeleteDialog: FC<TaskDeleteDialogProps> = ({
+	task,
+	onSuccess,
+	...props
+}) => {
+	const queryClient = new QueryClient();
+	const deleteTaskMutation = useMutation({
+		mutationFn: () =>
+			API.experimental.deleteTask(task.workspace.owner_name, task.workspace.id),
+		onSuccess: async () => {
+			await queryClient.invalidateQueries({ queryKey: ["tasks"] });
+		},
+	});
+
+	return (
+		<ConfirmDialog
+			{...props}
+			type="delete"
+			confirmLoading={deleteTaskMutation.isPending}
+			title="Delete task"
+			onConfirm={async () => {
+				try {
+					await deleteTaskMutation.mutateAsync();
+					displaySuccess("Task deleted successfully");
+					onSuccess?.();
+				} catch (error) {
+					displayError(
+						getErrorMessage(error, "Failed to delete task"),
+						getErrorDetail(error),
+					);
+				} finally {
+					props.onClose();
```

Comment:
Just curious if there is a reason `onSuccess` is being destructured above but not `onClose`.